support-models/src/main/scala/com/gu/support/catalog/Catalog.scala

```diff
@@ -20,54 +20,41 @@ object Catalog {
   } yield plan.id
 
   implicit val encoder: Encoder[Catalog] = deriveEncoder
-  implicit val decoder: Decoder[Catalog] = deriveDecoder[Catalog].prepare(mapFields)
 
-  private def mapFields(c: ACursor) = c.withFocus { json =>
-    val allRatePlans: Seq[Json] = json.\\("productRatePlans").flattenJsonArrays
-
-    val supportedRatePlans = allRatePlans
-      .filter(
-        _.getField("id")
-          .exists(id => productRatePlansWithPrices.exists(fromString(_) == id)),
-      )
-
-    val prices = supportedRatePlans.map { productRatePlan =>
-      val priceList = sumPriceLists(productRatePlan.\\("pricing"))
-      val id = productRatePlan.getField("id").getOrElse(Json.Null)
-      val saving = getSaving(productRatePlan)
-      Json.obj(
-        ("productRatePlanId", id),
-        ("savingVsRetail", saving),
-        ("prices", Json.fromValues(priceList)),
-      )
-    }
-    Json.obj(("prices", Json.fromValues(prices)))
-  }
-
-  def getSaving(json: Json) =
-    json
-      .getField("Saving__c")
-      .getOrElse(Json.fromString("null"))
-      .asString
-      .getOrElse("null") match {
-      case "null" => Json.Null
-      case i: String => Json.fromInt(i.toInt)
-    }
-
-  def sumPriceLists(priceLists: List[Json]): Iterable[Json] = {
+  def sumPriceLists(priceLists: Array[Price]): Array[(Price)] = {
     // Paper products such as Everyday are represented in the catalog as multiple
     // product rate plan charges (one for every day of the week) and these each
     // have their own price list. To get the total prices for these products therefore
     // we need to sum all of the price lists
-    priceLists.flattenJsonArrays
-      .flatMap(_.as[Price].toOption) // convert the Json to Price objects as they're easier to work with
+    priceLists
       .groupBy(_.currency)
-      .map(sumPrices)
-      .map({ case (_, price) => price.asJson }) // convert back to Json
+      .toArray
+      .map(sumPrices(_)._2)
```

Comment:
```suggestion
      .map(sumPrices)
      .map({ case (_, price) => price })
```
This is a nitpick, but to me this is easier to read because it shows that we are returning the price, `_2` could be anything

Reply:
Yep, agree easier to see price taken out. Changed.